### PR TITLE
Revamps project to use ES6 syntax.

### PR DESCRIPTION
**Pull request by glauco**
This commit does the following:
- Upgrades project dependencies
- Includes Snyk as a development dependency
- Includes handlebars as a development dependency
- Removes Mu2 which was not compatible with Node strict mode
- Includes Eslint checks
- Fixes lint errors

All of the while still making sure the example continues to work.

### CODE REVIEW

File: app/scripts/controllers/equipment.js [hunk 1]
```diff
@@ -1,102 +1,103 @@
...
- 	//init map
- 	$scope.map = {
- 		center:{
- 			latitude: 45,
- 			longitude: -73
- 		},
- 		zoom: 1,
- 		bounds: {}
- 	};
+function EquipmentController($routeParams, $scope, ApiService) {
```
Comment (bltavares) [inline app/scripts/controllers/equipment.js]: Is it recomnded to move controllers out of Angular when using ES6?

File: app/scripts/controllers/equipment.js [hunk 2]
```diff
@@ -1,102 +1,103 @@
...
+    geofenceAlarms.forEach((alarm, n) => {
+      const coordinates = alarm.alarm.geospatial.coordinates[0];
+      const paths = [];
+      coordinates.forEach((coordinate) => {
+        paths.push({
+          latitude: coordinate[1],
+          longitude: coordinate[0]
+        });
+      });
+      console.log(paths);
```
Comment (bltavares) [inline app/scripts/controllers/equipment.js]: I believe this `console.log` was left behind

File: app/scripts/controllers/main.js [hunk 3]
```diff
@@ -1,95 +1,85 @@
...
-		});
+    dealers.forEach((dealer) => {
+      dealerNames[dealer.id] = dealer.name;
+    });
 
-		//get engine hours
-		ApiService.getEngineHours().then(function(response) {
-			console.log('engine hours', response);
+    ApiService.getEngineHours().then((getEngineHoursResponse) => {
+      console.log('engine hours', getEngineHoursResponse);
```
Comment (bltavares) [inline app/scripts/controllers/main.js]: Should we be logging this response?

File: app/scripts/controllers/equipment.js (hunk 1 above)
Comment (glauco) [inline app/scripts/controllers/equipment.js]: No, it is not!

Just moved it to avoid the lint error complaining about an anonymous function.

File: app/scripts/controllers/equipment.js (hunk 2 above)
Comment (glauco) [inline app/scripts/controllers/equipment.js]: It was already in there! ;)

I can get rid of it, if needed.

File: app/scripts/controllers/main.js (hunk 3 above)
Comment (glauco) [inline app/scripts/controllers/main.js]: Given the goal was to only migrate to a newer and shinier syntax, I didn't touch anything related to the core functionality.

I could do that on a separated pull request. Would that be fine?

File: app/scripts/services/apiservice.js [hunk 4]
```diff
@@ -1,105 +1,130 @@
...
  * @description
  * # ApiService
  * Service in the fuseTestApp.
  */
- angular.module('fuseTestApp')
- .factory('ApiService', function (Config, $http) {
-
- 	$http.defaults.headers.common.Authorization = Config.BEARER_TOKEN;
+angular.module('fuseTestApp')
+  .factory('ApiService', (Config, $http) => {
```
Comment (bltavares) [inline app/scripts/services/apiservice.js]: Should the `ApiService` be extracted out of Angular as it was done to the controller?

Comment (glauco) [inline app/scripts/services/apiservice.js]: It seems that angular and arrow functions don't like each other.
It is not an issue for services, though.
What are your thoughts on it?
Should we standardise on extracting functions or should we use arrow functions were it is possible?

File: views/layouts/index.html [hunk 5]
```diff
@@ -33,18 +33,10 @@
...
+    <div ng-view></div>
     </div>
 
      <script>
-       !function(A,n,g,u,l,a,r){A.GoogleAnalyticsObject=l,A[l]=A[l]||function(){
-       (A[l].q=A[l].q||[]).push(arguments)},A[l].l=+new Date,a=n.createElement(g),
-       r=n.getElementsByTagName(g)[0],a.src=u,r.parentNode.insertBefore(a,r)
-       }(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
-
-       ga('create', 'UA-XXXXX-X');
```
Comment (bltavares) [inline views/layouts/index.html]: :+1:

File: index.js [hunk 6]
```diff
@@ -1,52 +1,37 @@
...
-var FUSE_TRACKERS_URL = process.env.FUSE_TRACKERS_URL || 'https://agco-fuse-trackers-sandbox.herokuapp.com';
-var FUSE_EM_URL = process.env.FUSE_EM_URL || 'http://fuse-em-api-dev.herokuapp.com';
-var OPENAM_USERNAME = process.env.OPENAM_USERNAME;
-var OPENAM_PASSWORD = process.env.OPENAM_PASSWORD;
-var OPENAM_URL = process.env.OPENAM_URL || 'https://aaat.agcocorp.com/auth/oauth2/access_token';
-var API_ID = process.env.API_ID;
-var API_SECRET = process.env.API_SECRET;
+const PORT = process.env.PORT || '8000';
+const FUSE_TRACKERS_URL = process.env.FUSE_TRACKERS_URL || 'https://agco-fuse-trackers-sandbox.herokuapp.com';
+const FUSE_EM_URL = process.env.FUSE_EM_URL || 'http://fuse-em-api-dev.herokuapp.com';
```
Comment (jeffhsta) [inline index.js]: the FUSE_EM_URL could be changed to use HTTPS on default link

File: index.js [hunk 7]
```diff
@@ -60,26 +45,24 @@ app.get('/', function(req, res) {
...
+      .then((response) => {
+        const parsedBody = JSON.parse(response);
+        const token = `${parsedBody.token_type} ${parsedBody.access_token}`;
+        return token;
+      });
+  }).then((token) => {
     res.render('index', {
-      FUSE_TRACKERS_URL: FUSE_TRACKERS_URL,
-      FUSE_EM_URL: FUSE_EM_URL,
+      FUSE_TRACKERS_URL,
```
Comment (jeffhsta) [inline index.js]: :+1:

File: app/scripts/controllers/equipment.js (hunk 2 above)
Comment (bltavares) [inline app/scripts/controllers/equipment.js]: It is ok then

File: app/scripts/controllers/main.js (hunk 3 above)
Comment (bltavares) [inline app/scripts/controllers/main.js]: Yup

File: app/scripts/services/apiservice.js (hunk 4 above)
Comment (bltavares) [inline app/scripts/services/apiservice.js]: I think that it would be interesting to make it more alike the Controller, but that could be on another PR

File: index.js (hunk 6 above)
Comment (glauco) [inline index.js]: It seems that it does not support HTTPS.

File: app/scripts/services/apiservice.js (hunk 4 above)
Comment (glauco) [inline app/scripts/services/apiservice.js]: I've just fixed it.

**Comment by bltavares** [comment]
:+1:
